src/gobprepare/importers/dump_importer.py

```diff
@@ -0,0 +1,107 @@
+import contextlib
+from io import StringIO
+import tempfile
+import gzip
+import re
+import tempfile
+from typing import List, Iterator, Optional
+
+from gobprepare.importers.typing import ReadConfig, SqlDumpImporterConfig
+from gobconfig.datastore.config import get_datastore_config
+from gobcore.datastore.factory import DatastoreFactory
+from gobcore.datastore.objectstore import ObjectDatastore
+from gobcore.datastore.sql import SqlDatastore
+from gobcore.logging.logger import logger
+
+from gobcore.exceptions import GOBException
+
+@contextlib.contextmanager
+def _load_from_objectstore(datastore: ObjectDatastore) -> Iterator[str]:
+    datastore.connect()
+
+    obj_info = next(datastore.query(None), None)
+
+    if obj_info is None:
+        raise GOBException(f"File not found on Objectstore: {datastore.read_config['file_filter']}")
+
+    _, obj = datastore.connection.get_object(
+        container=datastore.container_name, obj=obj_info["name"], resp_chunk_size=100_000_000
+    )
+
+    with tempfile.NamedTemporaryFile(mode="wb", suffix=".gz") as fp:
+        try:
+            for chunk in obj:
+                fp.write(chunk)
+            logger.info(f"File loaded from objectstore: {obj_info['name']}")
+            yield fp.name
+        finally:
+            datastore.disconnect()
+
+
+class SqlDumpImporter:
+    """Import a SQL dump into a SqlDatastore table."""
+
+    def __init__(self, dst_datastore: SqlDatastore, config: SqlDumpImporterConfig) -> None:
+        """ Initialise SqlDumpImporter """
+        self._dst_datastore = dst_datastore
+        # self._destination = config["destination"]
+        self._encoding = config.get("encoding", "utf-8")
+
+        self._read_config: Optional[ReadConfig] = config.get("read_config")
+        self._objectstore: Optional[str] = config.get("objectstore")
+        self._source: Optional[str] = config.get("source")
+
+    def _process_queries(self, queries: list) -> None:
```

Comment:
The logic in this method is a bit hard to follow. Can we rewrite this? Maybe also add an explanation of the different parameters or an example in a comment.

An idea: This method also does too much; it filters/rewrites queries and also executes them. Maybe split this into separate methods?

Reply:
Not done yet. In processing...

Reply:
It needs a lot of comments, but it's still unclear what's being done.

You're basically documenting what each line/part does, but you're not documenting the intention of the code. As a code reader I want to understand the intention.
Also; use proper method names and extract some fucntionality into functions/methods with apt names to remove the need for comments.

For example:
- On line 80, call that method _perform_query_substitutions() and the need for comments is gone. Then, add a docstring to the _perform_query_substituions to explain that it performs the substitutions as listed in read_config.
- Between lines 82 and 89, I'm not completely sure what's happening. On line 84 you're assigning the query to a variable copy_query, but that variable is somehow only read in the next iteration on line 88. Why? As this whole method revolves around this concept, that's the intention of this method and that should be documented in the docstring. If the docstring is clear and proper method/function names are used you can omit all of the other comments.

Reply:
Deels is verwerkt. 
In de Core heb ik een functie aangemaakt copy_expert() aanroept. Deze functie voert insert query uit. De functie neemt de copy query en de data als parameters, vandaar dat ik de insert query moet selecteren en bewaren en dan uitvoeren in de next iteratie met de data.
Een voorbeeld van de copy query en de data is:
```
COPY igp_pgplup_cmg_owner.kvkprsashm00 (ashid, functie, prsidh, prsidi, soort, prsashhibver) FROM stdin;
100000000007004957	Vennoot11111	100000000007004956	100000000007004958	OnbeperktBevoegd	0
100000000008718135	Vennoot22222	100000000008718134	100000000008718136	OnbeperktBevoegd	0
\.
```